array.go

```diff
@@ -913,3 +913,59 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Transform runs the passed function on all the Elements in the array.
+//
+// If transformation inside function fails, the original Array is marked failed.
+//
+// Transform will immediately stop execution of the function
+// upon first transformation failures
+// or if return value is nil.
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{"foo", "bar"})
+//	transformedArray := array.Transform(
+//		func(index int, value *httpexpect.Value) *httpexpect.Value {
+//			return *httpexpect.NewValue(...)
+//		},
+//	)
```

Comment:
Let's provide specific implementation of func in the example, and add one more line that demonstrated what is the result:

```
transformedArray.Equals([]interface{}{....})
```